### ISSUE TITLE
Target timeout when launching debug of Linux Program

### ISSUE THREAD

**Issue by kurt-wave** [mips-open-ide/Releases#17]
I installed the latest MIPS Open IDE on Windows 7.  The installer was downloaded from mipsopen.com.  I then created an Embedded Linux Hello World example program.

When I try to debug this, the QEMU window starts, but it never gets to a login prompt.  Then the debug launch fails with this error dialog:
![image](https://user-images.githubusercontent.com/48976304/55281394-1ab24f00-52f1-11e9-9c79-588a4e1a1487.png)

I'm guessing the timeout is because the QEMU target never finishes booting Linux?

This is a critical bug because you can't debug a Linux application.

However, I am able to create an Embedded Standalone program and debug it on the QEMU target.  The termination of the program in the debugger still gets to an infinite loop, but that issue is already documented in issue #14

**Comment by viosoft** [comment]
Depending on the speed of the host platform, Linux may take a while to boot on QEMU.   One may infer that your host has been well used, given that it runs Windows 7.   It is likely that the debugger attempts to download the application program and start debugging before QEMU is ready.

This scenario has been anticipated, and a workaround is provided by enabling the user to start Linux under QEMU separately before launching the debugger.  See: [https://github.com/mips-open-ide/Releases/wiki/Runtime-Environment](https://github.com/mips-open-ide/Releases/wiki/Runtime-Environment).  

With this flow, you can wait for the login prompt to appear before launching the debugger.   BTW:  This is a better way to go if you expect to debug the application repeatedly and not want to wait for Linux to boot each time.

**Comment by kurt-wave** [comment]
I did try to start QEMU separately, and that separate invocation also never gets to a command prompt.  I think there is something wrong with the Linux/QEMU combination in the current version.

I was able to use the previous version of the MIPS Open IDE (...0324) and the Linux debug works on this same PC. So I'm very sure the problem is not because the debugger tries to connect before QEMU is ready.  Well, that exactly the problem, except that QEMU is never ready.

**Comment by viosoft** [comment]
We'll have a look at Linux/QEMU on Windows 7 and follow up on this thread.

**Comment by kurt-wave** [comment]
Thanks for the response.  I did some more experimenting with my Windows 7 install of the MIPS Open IDE.  I started the QEMU window before launching the debug of my program.  I waited about 20 minutes, and the QEMU window never showed a login prompt, it just says "Guest has not initialized the display yet".

I also tried installing the latest MIPS Open IDE on Windows 10.  The QEMU window does get to a Linux login prompt.  And this QEMU works as the debug target for a Linux program.  So this problem does not occur for me on Windows 10, only on Windows 7.  I will wait to see what you find out.

**Comment by kurt-wave** [comment]
I'm still working on my Windows 7 PC.

I have continued to experiment with the MIPS Open IDE.  I tried creating new workspaces and new projects to see if I could get it to work.  I previously wrote that QEMU was "never ready".  I kept trying, but the QEMU window would never get to the login prompt.

Then at some point, the QEMU window got to the login prompt.   I'm not sure what I did to get it to work.  After that, I was able to debug my Linux program using the already started QEMU window.  Then, I closed that QEMU window and started a debug launch with QEMU closed.  This time the QEMU window that was started by the debug launch was able to get to the login prompt.  But the debug launch still timed out and failed.  So I left that QEMU window running and started the debug launch again.  This time the debugger was able to connect to the QEMU target and I could debug the program.

I kept trying over and over, and it seems to (somewhat randomly) go back and forth between the QEMU window never able to get to the login prompt, and then later the QEMU window would work again and I could debug my program.  Or sometimes the QEMU window gets to a login prompt but the debug launch still fails with a timeout.  Then it works the next time.  The thing is, I can't figure out how I'm making it fail sometimes, and then how I'm able to get it working again.  That's why I say it's somewhat random.

At this point, I can't tell if the boot of Linux on the QEMU target is unstable due to a bug in the MIPS Open IDE, or it is unstable due to something on my PC.  Maybe my hardware performance is not quite enough, or it is related to the tasks running in my system?  Is it possible that changing the debug launch timeout (is there one?) would fix this?  I didn't get the chance to try that, but I will see if I can try later.

I'm doing all of this with no other significant load on my PC, and during the debug launch the Windows Task Manager shows about 25% CPU for qemu-system-mips-exe and about 75% for the idle process.  My memory is also only half full.

So at least I know that this works on Windows 7, even though it does not work for me every time.
